### PR TITLE
Replace llvm disassembler with capstone

### PR DESCRIPTION
**Pull request by learn-more**
[capstone](https://www.capstone-engine.org/) is a library specifically built for this.

- Tested with WSL (linux build, instructions updated)
- Tested on msys2 (instructions updated)
- Tested on my MSVC port.

The disassembler uses intel syntax by default.
Unknown instructions are displayed as data (instead of halting the disassembly, called SKIPDATA mode)
I hope all disassembler architectures are configured correctly, I do not have samples available to confirm them all.
The code is formatted to look like the surrounding code. (Indented with tabs, empty lines also indented, no spaces around `(`, c-style comments, etc).

### CODE REVIEW

File: src/disassemble.cpp [hunk 1]
```diff
@@ -308,57 +269,47 @@ void REHex::Disassemble::update()
...
 	{
 		/* TODO: Report error */
 		return;
 	}
 	
-	/* Use Intel assembly syntax. */
-	LLVMSetDisasmOptions(disassembler, LLVMDisassembler_Option_AsmPrinterVariant);
+	/* Enable SKIPDATA mode */
+	/* This ensures the diassembler continues on unknown data, displaying it as '.byte 00' */
+	cs_option(disassembler, CS_OPT_SKIPDATA, CS_OPT_ON);
```
Comment (solemnwarning) [inline src/disassemble.cpp:288]: I think this option should probably be off - if you look in `REHex::Disassemble::update()` you'll see it attempts to align the disassembly with the cursor, or to the nearest point that produces valid disassembly, so you can click near an instruction and on certain architectures (almost any garbage looks like valid x86 machine code), it (might) find the instruction boundaries/alignment correctly.

Are there any situations you can think of where getting disassembly which is just full of "data" would be preferable?

**Comment by learn-more** [comment]
> Can confirm it seems to works fine on Debian Buster, will try to test it on OS X and give it a run with some different architectures' code soon(ish).

Awesome, thanks!
If you have data and the expected llvm output, I wouldn't mind adding that in a unittest.
It will most likely need re-formatting for whitespace / notation changes for capstone, but those can be easily verified.

File: src/disassemble.cpp (hunk 1 above)
Comment (learn-more) [reply src/disassemble.cpp:288]: Usually (at least on x64) it is just a one-byte pad:

image

But if you want this option disabled (which makes it behave more like it did before with llvm!) that is totally fine with me.

**Comment by solemnwarning** [comment]
Seems to work on OS X, specifying `CAPSTONE_SHARED=no` to prevent building the Capstone shared library.

Not too fussed on unit testing different architectures produce exactly the right assembly... I assume upstream's tests will cover that.

**Comment by learn-more** [comment]
> Seems to work on OS X, specifying `CAPSTONE_SHARED=no` to prevent building the Capstone shared library.
> 
> Not too fussed on unit testing different architectures produce exactly the right assembly... I assume upstream's tests will cover that.

That seems fair,
in that case manually verifying that capstone is configured correctly seems sufficient.

File: src/disassemble.cpp [hunk 2]
```diff
@@ -308,57 +269,43 @@ void REHex::Disassemble::update()
...
-	if(disassembler == NULL)
+	cs_err error = cs_open(desc.arch, desc.mode, &disassembler);
+	if(error != CS_ERR_OK)
 	{
 		/* TODO: Report error */
 		return;
 	}
-	
-	/* Use Intel assembly syntax. */
-	LLVMSetDisasmOptions(disassembler, LLVMDisassembler_Option_AsmPrinterVariant);
```
Comment (learn-more) [inline src/disassemble.cpp:328]: I removed the capstone skipdata mode

**Pull request merged**